pkg/kubelet/pod_workers.go

```diff
@@ -254,6 +274,11 @@ type podSyncStatus struct {
 	// to remove the pod. A terminal pod (Succeeded/Failed) will have
 	// termination status until the pod is deleted.
 	finished bool
+	// restartRequested is true if the pod worker was informed the pod is
+	// expected to exist (update type of create, update, or sync) after
+	// it has been killed. When known pods are synced, any pod that is
+	// terminated and has restartRequested will have its history cleared.
+	restartRequested bool
```

Comment:
i understand the approach taken here, i am wondering if its better to make config sources pod worker aware?    for example, could the config source peek at the list of pod workers and know there is a pod with a conflicting uid and wait at that moment to dispatch the HandlePodAddition until the pod worker was known terminal.